### ISSUE TITLE
Clarification on RLocalCachedMap

### ISSUE THREAD

**Issue by giantninja** [redisson/redisson#922]
Just looking for some clarification on how the RLocalCachedMap works.  

The documentation says it's better suited for read heavy operations (and less network round trips), which sounds perfect for something I'm implementing currently, which is a small map of runtime configurations (for simple settings like max for this and limit on that etc... Then the map in redis can be altered externally, and in turn my application without having to restart, since it's a service)

My question is does the RLocalCachedMap periodically update it's values from Redis every so often?  If so, how frequent?  If not, is there a mechanism to force it to update?  Would RLocalCachedMap be a good use case for having update on the fly, runtime config vals?

Thanks and keep up the awesome work!

P.S. 
Using Redisson 3.4.3 (sentinel master/slave setup), and Java 8 on Centos linux 64bit

**Comment by jackygurui** [comment]
For local cached map, there are two options you can configure to control the eviction and invalidation of the entries, please see below:

https://github.com/redisson/redisson/blob/master/redisson/src/main/java/org/redisson/api/LocalCachedMapOptions.java#L28

https://github.com/redisson/redisson/blob/master/redisson/src/main/java/org/redisson/api/LocalCachedMapOptions.java#L60

**Comment by giantninja** [comment]
So would it be true to say that a local cached map, that is essentially read only from Java/Redisson app (and updated externally/elsewhere in Redis), would update the values every 5 min or so? With a configuration something along these lines? (i.e. no eviction policy and maxIdle set to 5 minutes)

```
LocalCachedMapOptions.defaults()
    .evictionPolicy(LocalCachedMapOptions.EvictionPolicy.NONE)
    .cacheSize(10)
    .maxIdle(5, TimeUnit.MINUTES);
```

**Comment by mrniko** [comment]
> My question is does the RLocalCachedMap periodically update it's values from Redis every so often?

Actually, Redisson doesn't update values itself it's just invalidate them. Invalidate in realtime thanks to pubsub channel used inside. I agree that update policy would be useful too. I'll think how to implement it.

**Comment by giantninja** [comment]
Ok, thanks for the clarification @mrniko .  I figured it may or may not work that way, but I agree that it might be a useful feature.

Just one last question on this before closing the ticket.  What is the difference between `timeToLive()` and `maxIdle()` ?  The former seems obvious, but the latter is what I figured was the time it lived in local map memory before checking redis again (if no eviction policy), but since you've confirmed that functionality hasn't been added yet, what's that setting for?

Thanks again!
